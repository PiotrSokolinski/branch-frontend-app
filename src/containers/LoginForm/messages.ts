import { defineMessages } from "react-intl";

export const scope = "app.containers.LoginForm";

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: "Don’t miss your next favorite.",
  },
  emailPlaceholder: {
    id: `${scope}.emailPlaceholder`,
    defaultMessage: "Address e-mail",
  },
  emailLabel: {
    id: `${scope}.emailLabel`,
    defaultMessage: "E-mail",
  },
  passwordPlaceholder: {
    id: `${scope}.passwordPlaceholder`,
    defaultMessage: "Password",
  },
  passwordLabel: {
    id: `${scope}.passwordLabel`,
    defaultMessage: "Password",
  },
  signIn: {
    id: `${scope}.signIn`,
    defaultMessage: "Sign in",
  },
  register: {
    id: `${scope}.register`,
    defaultMessage: "Don’t have an account yet?",
  },
  emailInvalidError: {
    id: `${scope}.emailInvalidError`,
    defaultMessage: "E-mail address is invalid",
  },
  emailEmptyError: {
    id: `${scope}.emailEmptyError`,
    defaultMessage: "E-mail address is required",
  },
  passwordEmptyError: {
    id: `${scope}.passwordEmptyError`,
    defaultMessage: "Password is required",
  },
});
