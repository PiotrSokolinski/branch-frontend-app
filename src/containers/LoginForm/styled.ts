import styled from "styled-components";

import ButtonLink from "../../components/ButtonLink";
import Button from "../../components/Button";
import Error from "../../components/Error";

export const StyledError = styled(Error)`
  margin-top: 2rem;
  font-size: 1.3rem;
  align-self: center;
`;

export const Container = styled.div`
  display: flex;
  margin-top: 10rem;
`;

export const Form = styled.form`
  display: flex;
  width: 40rem;
  flex-direction: column;
`;

export const Wrapper = styled.div`
  margin-bottom: 2rem;
`;

export const StyledButtonLink = styled(ButtonLink)`
  align-self: center;
`;

export const StyledButton = styled(Button)`
  width: 100%;
  height: 5rem;
  font-size: 2rem;
  margin-top: 2rem;
  margin-bottom: 2rem;
`;

export const Header = styled.h1`
  font-size: 4rem;
  margin-bottom: 3rem;
`;
