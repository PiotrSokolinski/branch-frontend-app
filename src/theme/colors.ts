const colors = {
  aquaHaze: "#f4f7f9",
  aquaHazeDark: "#E9EEF3",
  white: "#FFFFFF",
  black: "#000000",
  coralLight: "#FFE7DD",
  coralDark: "#FF7C44",
  java: "#20BFB7",
  javaLight: "#D2F2F1",
  javaDark: "#1CB9B0",
  blueDianne: "#21455B",
  hawkesBlue: "#EDF4FE",
  pigeonPost: "#AECBDD",
  flamingo: "#ef462f",
  tarawera: "#0A354F",
  boulder: "#757575",
  downriver: "#0B2B4A",
  nileBlue: "#1A475F",
  frenchGray: "#BABDC8",
  athensGray: "#E3E6EC",
  zumthor: "#E8F1FF",
  zircon: "#F5F7F9",
  solitude: "#E1EDFF",
  coral: "#FE8856",
  lightGray: "#D3D3D3",
  athensGrayLight: "#F6F7F9",
  bermudaGray: "#6C8B9F",
  tropicalBlue: "#CFDFF9",
  jungleGreen: "#2aaba4",
  linkWater: "#DEE8F5",
  lynch: "#5B7585",
  gray: "#808080",
  iron: "#D0D2D3",
  endeavour: "#005DAA",
  pink: "#DE138B",
  mystic: "#E8EDF2",
  mentation: "#FFA336",
  medication: "#4356FF",
  mobility: "#FA4AE8",
  geyser: "#D3DCDF",
  baliHai: "#90A2AD",
  milanoRed: "#AD1B07",
  EasternBlue: "#1DB0A8",
  viking: "#59DED7",
};

export default colors;
