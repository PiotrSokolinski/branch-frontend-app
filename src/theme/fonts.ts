const fonts = {
  h1: {
    fontFamily: "Poppins",
    fontWeight: "bold",
    fontSize: "5rem",
    lineHeight: "2.7rem",
    letterSpacing: "0.08em",
  },
  h2: {
    fontFamily: "Poppins",
    fontWeight: "normal",
    fontSize: "3rem",
    lineHeight: "1.9rem",
    letterSpacing: "0.08em",
  },
  text: {
    fontFamily: "Poppins",
    fontWeight: "normal",
    fontSize: "1.6rem",
    lineHeight: "1.4rem",
    letterSpacing: "0.08em",
  },
  button: {
    fontFamily: "Poppins",
    fontWeight: "bold",
    fontSize: "1.3rem",
    lineHeight: "1.4rem",
    letterSpacing: "0.08em",
  },
  label: {
    fontFamily: "Poppins",
    fontWeight: "bold",
    fontSize: "1.3rem",
    lineHeight: "1.4rem",
    letterSpacing: "0.02em",
  },
  p: {
    fontFamily: "Poppins",
    fontWeight: "normal",
    fontSize: "1.1rem",
    lineHeight: "1.3rem",
    letterSpacing: "0.02em",
  },
  input: {
    fontFamily: "Poppins",
    fontWeight: "normal",
    fontSize: "1.3rem",
    lineHeight: "1.4rem",
    letterSpacing: "0.08em",
  },
};

export default fonts;
