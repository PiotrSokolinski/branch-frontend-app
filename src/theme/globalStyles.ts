import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  html,
  body {
    font-family: Poppins !important;
    margin: 0;
    padding: 0;
    font-size: 1.3rem;

  }

  * {
    box-sizing: border-box;
  }

  html {
    height: 100%;
    width: 100%;
  }

  body {
    min-height: 100%;
    max-width: 100%;
    display: flex;
    flex: 1;
    align-items: stretch;
    overflow-y: scroll;
    overflow-x: hidden;
  }

  h1 { margin: 0; font-size: 2.1rem; font-weight: bold; }
  h2 { margin: 0; font-size: 1.8rem; font-weight: bold; }
  p { margin: 0; }


  #root {
    display: flex;
    min-height: 100%;
    min-width: 100%;
  }
`;

export default GlobalStyles;
