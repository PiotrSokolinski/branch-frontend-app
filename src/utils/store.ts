import axios from "axios";
import thunk from "redux-thunk";
import { createStore, applyMiddleware, combineReducers } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";

import history from "./history";
import { registerReducer } from "../redux/register";
import { loginReducer } from "../redux/login";
import { productsReducer } from "../redux/products";
import { ordersReducer } from "../redux/orders";

const reducer = combineReducers({
  register: registerReducer,
  login: loginReducer,
  orders: ordersReducer,
  products: productsReducer,
});

const store = createStore(
  reducer,
  composeWithDevTools(
    applyMiddleware(thunk.withExtraArgument({ axios, history }))
  )
);

export default store;
