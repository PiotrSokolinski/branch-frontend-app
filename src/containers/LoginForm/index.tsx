import React from "react";
import * as Yup from "yup";
import { useFormik, FormikHelpers } from "formik";
import { FormattedMessage, useIntl } from "react-intl";

import {
  Form,
  Wrapper,
  Container,
  StyledButtonLink,
  StyledButton,
  Header,
} from "./styled";
import Input from "../../components/Input";
import messages from "./messages";
import FormField from "../../components/FormField";
import { useDispatch, useSelector } from "react-redux";
import { login, selectError, selectLoading } from "../../redux/login";
import { StyledError } from "../RegisterForm/styled";

type LoginFormProps = {};

type Inputs = {
  email: string;
  password: string;
};

const initialValues: Inputs = {
  email: "",
  password: "",
};

const validationSchema = (formatMessage: any) =>
  Yup.object().shape({
    email: Yup.string()
      .email(formatMessage(messages.emailInvalidError))
      .required(formatMessage(messages.emailEmptyError)),
    password: Yup.string().required(formatMessage(messages.passwordEmptyError)),
  });

const LoginForm: React.FC<LoginFormProps> = () => {
  const dispatch: any = useDispatch();
  const loading: boolean = useSelector(selectLoading);
  const error: string = useSelector(selectError);
  const { formatMessage } = useIntl();
  const onSubmit = async (
    { email, password }: Inputs,
    { setSubmitting }: FormikHelpers<Inputs>
  ) => {
    dispatch(login(email, password));
    setSubmitting(false);
  };

  const schema: Yup.ObjectSchema = validationSchema(formatMessage);
  const {
    handleSubmit,
    values,
    errors,
    touched,
    handleChange,
    handleBlur,
  } = useFormik({
    initialValues,
    onSubmit,
    validationSchema: schema,
  });

  return (
    <Container>
      <Form onSubmit={handleSubmit}>
        <Header>Sign in</Header>
        <Wrapper>
          <FormField
            error={touched.email && errors.email}
            label={formatMessage(messages.emailLabel)}
            name="email"
          >
            <Input
              id="email"
              type="text"
              name="email"
              value={values.email}
              onChange={handleChange}
              onBlur={handleBlur}
              placeholder={formatMessage(messages.emailPlaceholder)}
            />
          </FormField>
        </Wrapper>
        <Wrapper>
          <FormField
            error={touched.password && errors.password}
            label={formatMessage(messages.passwordLabel)}
            name="password"
          >
            <Input
              id="password"
              type="password"
              name="password"
              value={values.password}
              onChange={handleChange}
              onBlur={handleBlur}
              placeholder={formatMessage(messages.passwordPlaceholder)}
            />
          </FormField>
        </Wrapper>
        <StyledButton type="submit" loading={loading} disabled={loading}>
          <FormattedMessage {...messages.signIn} />
        </StyledButton>
        <StyledButtonLink to="/register">
          <FormattedMessage {...messages.register} />
        </StyledButtonLink>
        {error && <StyledError>{error}</StyledError>}
      </Form>
    </Container>
  );
};

export default LoginForm;
