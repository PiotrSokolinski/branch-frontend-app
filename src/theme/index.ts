import { DefaultTheme } from "styled-components";

import colors from "./colors";
import fonts from "./fonts";
import zIndex from "./zIndex";
import layouts from "./layouts";

import { rem } from "./helpers";

const theme: DefaultTheme = {
  colors,
  fonts,
  zIndex,
  layouts,
  rem,
};

export default theme;
