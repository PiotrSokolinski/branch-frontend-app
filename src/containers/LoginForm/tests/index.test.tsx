/**
 *
 * Tests for Login
 *
 */

import React from "react";
import { Provider } from "react-redux";
import { IntlProvider } from "react-intl";
import { MemoryRouter } from "react-router-dom";
import { render, fireEvent } from "@testing-library/react";

import store from "../../../utils/store";

import LoginForm from "../index";
import { ThemeProvider } from "styled-components";
import theme from "../../../theme";

describe("<LoginForm />", () => {
  it("Expect to not log errors in console", () => {
    const spy: jest.SpyInstance<
      void,
      [message?: any, ...optionalParams: any[]]
    > = jest.spyOn(global.console, "error");
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ThemeProvider theme={theme}>
            <MemoryRouter>
              <LoginForm />
            </MemoryRouter>
          </ThemeProvider>
        </IntlProvider>
      </Provider>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it("Should render primary button and match the snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ThemeProvider theme={theme}>
            <MemoryRouter>
              <LoginForm />
            </MemoryRouter>
          </ThemeProvider>
        </IntlProvider>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("Should invoke onClick callback on the Button", () => {
    const { getAllByText, getByPlaceholderText } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ThemeProvider theme={theme}>
            <MemoryRouter>
              <LoginForm />
            </MemoryRouter>
          </ThemeProvider>
        </IntlProvider>
      </Provider>
    );
    const emailInput: HTMLElement = getByPlaceholderText("Address e-mail");
    const passwordInput: HTMLElement = getByPlaceholderText("Password");
    const button: HTMLElement = getAllByText("Sign in")[1];

    fireEvent.change(emailInput, { value: "email@gmail.com" });
    fireEvent.change(passwordInput, { value: "Password1!" });
    fireEvent.click(button);

    expect(store.getState().login.loading).toEqual(false);
  });
});
