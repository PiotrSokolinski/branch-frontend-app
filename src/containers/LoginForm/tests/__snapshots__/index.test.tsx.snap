// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> Should render primary button and match the snapshot 1`] = `
<div>
  <div
    class="sc-jSgupP hiHddL"
  >
    <form
      class="sc-gKsewC eVVmvP"
    >
      <h1
        class="sc-jrAGrp uXCsa"
      >
        Sign in
      </h1>
      <div
        class="sc-iBPRYJ fIgKZv"
      >
        <div
          class="sc-crrsfI jntZes"
        >
          <label
            class="sc-iqHYGH hUVMec"
            for="email"
          >
            E-mail
          </label>
          <input
            class="sc-kEjbxe iRwQfK"
            id="email"
            name="email"
            placeholder="Address e-mail"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-iBPRYJ fIgKZv"
      >
        <div
          class="sc-crrsfI jntZes"
        >
          <label
            class="sc-iqHYGH hUVMec"
            for="password"
          >
            Password
          </label>
          <input
            class="sc-kEjbxe iRwQfK"
            id="password"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        class="sc-dlfnbm gfORQY sc-pFZIQ dqXbvk"
        type="submit"
      >
        Sign in
      </button>
      <a
        class="sc-bdfBwQ uLwuk sc-fubCfw komXDx"
        href="/register"
      >
        Donâ€™t have an account yet?
      </a>
    </form>
  </div>
</div>
`;
